### ISSUE TITLE
[Suggestion] Typed exception for invalid object names

### ISSUE THREAD

**Issue by ig-sinicyn** [minio/minio-dotnet#395]
MinioClient v3.1.11

**Current behavior:**
requests with invalid object names throw InternalClientException, as example
```cs
var blobName = new string('A', 260);
await MinioClient.GetObjectAsync(
	bucketName,
	blobName,
	s => { ... });
```

**Expected behavior:**
Code should throw InvalidObjectNameException.

**Comment by kannappanr** [comment]
@ig-sinicyn Please feel free to send a PR

**Comment by ig-sinicyn** [comment]
Done, #398

**Comment by poornas** [comment]
Fixed by #398
